src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,248 @@
 package infrastructure.content;
 
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ModuleID;
+import java.util.HashMap;
+import networking.CommunicatorFactory;
+import networking.ICommunicator;
+import networking.INotificationHandler;
+import org.json.JSONObject;
+
+/**
+ * This class ContentCommunicator implements the IContentCommunicator interface and all the 
+ * methods inside it. The ideal purpose of this class is to initialise any new client, send 
+ * message from UI to networking module, handle client exit with respect to content module
+ * and handle the subscribers of content module.
+ * 
+ * @author Badal Kumar (111701008)
+ */
 public class ContentCommunicator implements IContentCommunicator{
-    
+	/**
+	 * This variable will store the username/userid of client as a String.
+	 */
+	private static String userName;
+	
+	/**
+	 * This variable will store the image of client in String format.
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String.
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private static HashMap<String, String> imageMap = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module,
+	 * identifier is of type String while handler is of type IContentNotificationHandler.
+	 */
+	private static HashMap<String, IContentNotificationHandler> handlerMap = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * An instance of ServerPort class is created and stored the variable serverPort.
+	 */
+	private ServerPort serverPort = new ServerPort();

```

Comment:
This should be from the factory as well

Reply:
But then I can't use getPort() method as it is not there in IServerPort interface.

Reply:
I have made changes to IServerPort interface, added the getPort() method. Now in CC, serverPort is obtained from ContentFactory class.

Reply:
I don't understand what you did. Did you change the interface? That shouldn't be done unless there is no other way

Reply:
You were using this getPort() even before adding ServerPort to factory, right. You didn't have it in the interface?

Reply:
yes